misc. changes (builder, docs, installer)

topic2k opened the pull request:
- small changes to the docs sources and builder
- a little cosmetics for the installer and the app icon
- prepare the build script for uploading docs to website

kdschlosser commented on _build/builder/BuildDocs.py:
```diff
@@ -116,22 +123,24 @@ def Setup(self):
+        )
+
+
+def BuildClsDocs(clsNames, doc_src_dir):
```
I am thinking we should not include a list of plugins in the docs. as these are external pieces of code that technically speaking not part of the core of EG and are supposed to be maintained by a 3rd party. people think that because it is included in the core that they are maintained by us. Which on a side not we should change the authors to any of the ones that are maintained or written by the core team. to "The EventGhost Team" This can be changed in the future I guess. I do not even know if the list is correct to the plugins that are shipped with EG. I would have to check.

kdschlosser commented on _build/builder/BuildDocs.py:
```diff
@@ -116,22 +123,24 @@ def Setup(self):
+            [
+                "-b", "html",
+                "-D", "version=%s" % self.buildSetup.appVersion,
+
```
blank line?

topic2k replied:
uups, where are my glasses 👓  :D

kdschlosser commented on _build/builder/BuildDocs.py:
```diff
@@ -116,22 +123,24 @@ def Setup(self):
+                "-b", "html",
+                "-D", "version=%s" % self.buildSetup.appVersion,
+
+
```
blank line?

kdschlosser commented on _build/builder/BuildDocs.py:
```diff
@@ -116,22 +123,24 @@ def Setup(self):
+                join(self.buildSetup.websiteDir, "docs"),
+            ]
+        )
+
```
almost duplicate code to lines 93-103 any way to condense this?

topic2k replied:
replaced with a common function

kdschlosser commented on _build/builder/BuildDocs.py:
```diff
@@ -73,27 +88,19 @@ def Setup(self):
-            EncodePath(DOCS_SOURCE_DIR),
-            tmpDir,
-        ])
+        WritePluginList(join(self.buildSetup.docsDir, "pluginlist.rst"))
```
I am thinking we should not include a list of plugins in the docs. as these are external pieces of code that technically speaking not part of the core of EG and are supposed to be maintained by a 3rd party. people think that because it is included in the core that they are maintained by us. Which on a side not we should change the authors to any of the ones that are maintained or written by the core team. to "The EventGhost Team" This can be changed in the future I guess. I do not even know if the list is correct to the plugins that are shipped with EG. I would have to check.

per1234 replied:
>I am thinking we should not include a list of plugins in the docs

 I'm not super happy with having one[ plugin list in the docs](http://www.eventghost.net/docs/pluginlist.html) (the bundled plugins) and another[ list in the wiki](http://www.eventghost.net/mediawiki/index.php?title=Plugins) (bundled plugins and 3rd party plugins) but since we already have a plan to make both of those lists obsolete ([plugin repository](https://github.com/EventGhost/EventGhost/issues/4)) it's not something that concerns me much. I'm fine with leaving the system as is for now. I think the energy is better spent on the long term goal of the plugin repository rather than a minor issue that will solve itself in time, possibly before the updated docs even hit the website.

> I do not even know if the list is correct to the plugins that are shipped with EG. I would have to check.

Isn't the plugin list automatically generated? If so it should be complete, with the exception of the plugins that have the `noinclude` flag file.

kdschlosser replied:
the reason why i do not know if it is complete or not is I do not know if it is automatically generated. because there is a "pluginlist.rst" file. if it was something that was automatically generated would there be an rst file for the plugins? the docstrings are automatically generated for specific functions/methods in EG and those do not have an rst file.

per1234 replied:
I think it is generated here: https://github.com/topic2k/EventGhost/blob/a04c08da1ea7352fa3cf886293445f595964bed1/_build/builder/BuildDocs.py#L197-L250

kdschlosser replied:
Oh yeah. and when i stated This can be changed in the future I guess. I was talking about this piece in general and not the whole thing about the authors. my bad on that one. i just read over my comment again and I kind of lost myself with that one as well. 

LOL

it's good to note it as we come across different bits of code that may have to be changed even if it is in the future. it. I do not know if there is a way to search through the comments for specific phrases. or maybe we could place a marker in the comment that way we have the ability when the time comes to make the changes in that specific area of code for the new addition. and or change. this is a good way to make sure we "remember" all of the things we need to modify.

kdschlosser replied:
OK i see what it is doing. it is creating the rst file.

topic2k replied:
The plugin list have always been there, it's automatically build. I left there but would remove it once we have a plugin manager.

kdschlosser replied:
ok cool. just wanted to make a note of it is all. so when the time comes we know we need to remove it... my way of making a mental impression

topic2k commented:
> I have not tried to use this as it would fail appveyor. But it looks good to me.

I see no errors on appveyor. Is it on your own account? I've updated sphinx version and added paramiko, so the cache needs to be invalidated/deleted.

kdschlosser commented:
no no what I am saying is.. if there was an error in the code. it would fail the appveyor build. and it didn't. so the code should be good.
